holoviews/operation/datashader.py

```diff
@@ -468,6 +458,27 @@ def _process(self, element, key=None):
             vdims = Dimension('Count')
         params['vdims'] = vdims
 
+        if x is None or y is None or width == 0 or height == 0:
+            xarray = xr.DataArray(np.full((height, width), np.NaN, dtype=np.float32),
```

Comment:
Why does it have to be ``np.float32`` in these two cases?

Reply:
No good reason, pointless optimization. Can take it out.

Reply:
I wouldn't specify a precision unless there is a good reason.